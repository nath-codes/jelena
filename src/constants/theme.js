export default {
  fonts: {
    heading: "bebas-neue",
    body: "montserrat"
  },
  colors: {
    delftBlue: "#1b56c5",
  }
};
