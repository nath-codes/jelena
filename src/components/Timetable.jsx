import React from "react";
import styled from "styled-components";
import spacer from "../mixins/spacer";
import Heading from "./Heading";
import Section from "./Section";
import List from "./List";
import timetable from "../constants/timetable";
import { medium } from "../constants/breakpoints";

const TimetableList = List.extend`
  @media (min-width: ${medium}) {
    margin-left: ${props => spacer(6.5, props.theme.breakpoint)};
  }
`;

const TimetableSection = Section.extend`
  padding-bottom: 0;
`;

const TimetableItem = styled.li`
  align-items: center;
  display: flex;
  flex-direction: column;
  font-size: ${props => spacer(0.75, props.theme.breakpoint)};
  margin-bottom: ${props => spacer(2, props.theme.breakpoint)};
  text-align: center;

  &:last-child {
    margin-bottom: 0;
  }

  .time {
    color: ${props => props.theme.colors.delftBlue};
    margin-bottom: ${props => spacer(1, props.theme.breakpoint)};
    min-width: ${props => spacer(4.5, props.theme.breakpoint)};
    white-space: nowrap;
  }

  @media (min-width: ${medium}) {
    flex-direction: row;
    margin-bottom: ${props => spacer(1.5, props.theme.breakpoint)};
    text-align: left;

    .time {
      margin-bottom: 0;
      white-space: nowrap;
    }
  }
`;

const TimeTable = () => (
  <TimetableSection id="timetable" tight>
    <Heading>Timetable</Heading>
    <TimetableList>
      {timetable.map(item => (
        <TimetableItem key={item.time}>
          <div className="time">{item.time}</div>
          <div className="title">{item.title}</div>
        </TimetableItem>
      ))}
    </TimetableList>
  </TimetableSection>
);

export default TimeTable;
