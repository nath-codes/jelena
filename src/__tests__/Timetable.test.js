import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";
import Timetable from "../components/Timetable";
import theme from "../constants/theme";

describe("<Timetable />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Timetable />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
