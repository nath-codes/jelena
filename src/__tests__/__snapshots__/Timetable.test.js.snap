// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timetable /> renders correctly 1`] = `
.c1 {
  color: #1b56c5;
  font-family: bebas-neue;
  font-size: 3.5999999999999996rem;
  font-weight: 400;
  line-height: 3.5999999999999996rem;
  margin: 0 0 2.4rem;
  text-align: center;
}

.c2 {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  margin: 0 auto;
  padding: 2.4rem 0;
  position: relative;
  width: 80%;
  padding-bottom: 0;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.8999999999999999rem;
  margin-bottom: 2.4rem;
  text-align: center;
}

.c3:last-child {
  margin-bottom: 0;
}

.c3 .time {
  color: #1b56c5;
  margin-bottom: 1.2rem;
  min-width: 5.3999999999999995rem;
  white-space: nowrap;
}

@media (min-width:768px) {
  .c2 {
    margin-left: 7.8rem;
  }
}

@media (min-width:768px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-bottom: 1.7999999999999998rem;
    text-align: left;
  }

  .c3 .time {
    margin-bottom: 0;
    white-space: nowrap;
  }
}

<section
  className="c0"
  id="timetable"
>
  <h2
    className="c1"
  >
    Timetable
  </h2>
  <ul
    className="c2"
  >
    <li
      className="c3"
    >
      <div
        className="time"
      >
        1:30 pm
      </div>
      <div
        className="title"
      >
        Arrive
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        2:00 pm
      </div>
      <div
        className="title"
      >
        Ceremony
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        2:30 pm
      </div>
      <div
        className="title"
      >
        Reception drinks
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        3:45 pm
      </div>
      <div
        className="title"
      >
        Wedding breakfast
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        5:45 pm
      </div>
      <div
        className="title"
      >
        Speeches
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        7:00 pm
      </div>
      <div
        className="title"
      >
        First dance
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        8:30 pm
      </div>
      <div
        className="title"
      >
        Cutting of the (cheese) cake
      </div>
    </li>
    <li
      className="c3"
    >
      <div
        className="time"
      >
        Midnight
      </div>
      <div
        className="title"
      >
        Carriages
      </div>
    </li>
  </ul>
</section>
`;
